src/util/campaignUtils.js

```diff
@@ -0,0 +1,49 @@
+import gql from 'graphql-tag';
+
+function addCreditByType(type, code, apollo) {
+	const addCreditMutation = gql`mutation addCreditMutation(
+		$basketId: String!,
+		$creditType: CreditTypeEnum!,
+		$redemptionCode: String!
+	) {
+		shop(basketId: $basketId) {
+			addCreditByType(creditType: $creditType, redemptionCode: $redemptionCode)
+		}
+	}`;
+	return new Promise(() => {
```

Comment:
You can simplify all these by just returning apollo.mutate, since it itself is a promise

Reply:
ditto